### PR TITLE
Add supportive

### PR DESCRIPTION
**Pull request by sriking1783**
This is supportive

### CODE REVIEW

File: supportive/supportive.rb [hunk 1]
```diff
@@ -9,92 +9,57 @@ class BlagPost
...
 
   private
 
   def byline
-    if author.nil?
-      ""
-    else
-      "By #{author.name}, at #{author.url}"
-    end
+    author.nil? ? "" : "By #{author.name}, at #{author.url}"
```
Comment (jmmastey) [inline supportive/supportive.rb]: Take a look at `.try`

File: supportive/supportive.rb [hunk 2]
```diff
@@ -9,92 +9,57 @@ class BlagPost
...
-      when 3 then "rd"
-      else "th"
-    end
-    "You will be the #{comments.length}#{ordinal} commenter"
+    "You will be the #{commenter_number} commenter"
   end
 
   def comments_allowed?
-    publish_date + (365 * 3) > Date.today
+    publish_date > 3.years.ago.to_date
```
Comment (jmmastey) [inline supportive/supportive.rb]: the `to_date` is superfluous, but otherwise good

File: supportive/supportive.rb [hunk 3]
```diff
@@ -9,92 +9,57 @@ class BlagPost
...
   end
 
   def abstract
-    if body.length < 200
-      body
-    else
-      body[0..200] + "..."
-    end
+    return body.truncate(204) if body.length >= 200
+    body
```
Comment (jmmastey) [inline supportive/supportive.rb]: Truncate will do the right thing if the body is too long. You can turn the entire block into `body.truncate(200)`

**Comment by jmmastey** [comment]
Added a note or two. Looks good!

File: supportive/supportive.rb (hunk 3 above)
Comment (sriking1783) [inline supportive/supportive.rb]: I did 204, to get the effect of the original one. If I do 200, it get 197 characters and adds 3 dots at the end.
So, I did 204, :)

**Pull request closed without merging**